### PR TITLE
Remove else if and put early return on choose handler

### PR DESCRIPTION
**Pull request by Calison25**
I was pass for a problem with guzzle, and try to figure out was doing and find this method, so I think he can be a little better, and remove else if and put early return for make it more readable and faster.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: '3.6'
```
Comment (GrahamCampbell) [inline docker-compose.yml:1]: Please don't add this unrelated change.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by GrahamCampbell** [comment]
What does this fix, please? :)

File: docker-compose.yml (hunk 1 above)
Comment (Calison25) [reply docker-compose.yml]: sorry, my mistake with docker-compose.

**Comment by Calison25** [comment]
> What does this fix, please? :)

Nothing, only make method more readable, with early return, I don't find bug in this method, only try to make more readable.

File: src/Utils.php [hunk 2]
```diff
@@ -90,26 +90,38 @@ public static function debugResource($value = null)
...
-        } elseif (!$handler) {
+        if (!$handler) {
             throw new \RuntimeException('GuzzleHttp requires cURL, the allow_url_fopen ini setting, or a custom HTTP handler.');
         }
 
         return $handler;
     }
 
+    /**
+     * @return callable|CurlHandler|CurlMultiHandler|null
```
Comment (GrahamCampbell) [inline src/Utils.php:106]: wrong phpdoc. should be `(callable(\Psr\Http\Message\RequestInterface, array): \GuzzleHttp\Promise\PromiseInterface)|null`

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Utils.php [hunk 3]
```diff
@@ -90,26 +90,38 @@ public static function debugResource($value = null)
...
+        if (\ini_get('allow_url_fopen')) {
+            return $handler ? Proxy::wrapStreaming($handler, new StreamHandler()) : new StreamHandler();
         }
 
-        if (\ini_get('allow_url_fopen')) {
-            $handler = $handler
-                ? Proxy::wrapStreaming($handler, new StreamHandler())
-                : new StreamHandler();
-        } elseif (!$handler) {
+        if (!$handler) {
```
Comment (GrahamCampbell) [inline src/Utils.php:98]: `$handler === null`

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Utils.php (hunk 2 above)
Comment (Calison25) [reply src/Utils.php]: Changed, it was generated automatically by PHP storm.

**Comment by GrahamCampbell** [comment]
I am not convinced this makes it easier to read, actually. Now is a bit confusing, since the curl handler is the only thing returned from getHandler, and the main method possibly builds a stream handler. I think it was better before. Since this is not broken, I'm going to close this. Thanks anyway. :)

File: src/Utils.php (hunk 3 above)
Comment (Calison25) [reply src/Utils.php]: changed too.
